rust/src/lib.rs

```diff
@@ -200,6 +200,11 @@ pub mod http {
             headers.insert(header, value.to_string());
         }
 
+        let redirect = headers.get("location").map(|url| request(url));
+        if redirect.is_some() {
+            return redirect.unwrap();
+        }
```

Comment:
```suggestion
        if let Some(redirect) = headers.get("location").map(request) {
            return redirect;
        }
```

Reply:
![image](https://user-images.githubusercontent.com/71239005/116818692-a91bed80-aba7-11eb-8de3-661e4c827374.png)

Reply:
Then, `map(|url| request(url)` please. The major goal is removnig unnessessary `is_some()` and `unwrap()`

Reply:
I now think request part must be in the body of the if block because request is a failable function in theory - though we are not handling it here. So a new suggestion is:
```rust
        if let Some(url) = headers.get("location") {
            return request(url);
        }
```

this is also equivalent to the python version.